## 🏡 CFG +Masters Project: Airbnb Investment Analysis in Essex

This project dives into the world of **Airbnb listings in Essex** with the goal of assisting a holiday lettings company select profitable properties to invest in based on historical Airbnb listing data. 


### 🧠 Project Objective

A holiday lettings company is on the lookout for promising properties across Essex. They are seeking data insights to select properties that can earn the highest income. 

The aims of this project are:
- identify the features that mostly correlates with occupancy rates and annual revenue
- build a machine learning model that predicts which types of properties achieves the highest occupancy rates and revenue potential. 

### 📈 The Data

The data is retrieved from real data acquired from PriceLabs, a subscription-based database that provides professionally managed datasets for existing and potential Airbnb businesses.

### 🛠️ Phase 1 : Data Wrangling and Exploratory Data Analysis

In this first assignment, I am focussing on:
 - **Merging and Data splitting**: to enable a comprenshensive data set, this individual sheets are first merged, and then split into training, validation and testing datasets. 
- **Data cleaning**: To manage missing data, fixing outliers, handling duplicates etc.
- **Pre-processing and transformation**: Encoding and transforming columns as needed.
- ** Exploratory Data Analysis (EDA)**


### 🔍 What I'm Exploring

- **Host features**: Reviews, professionally managed
- **Property Details**: Property type, number of bathrooms, number of occupants 
- **Amenities**: seafront, hot-tub, pets allowed etc
- **Listing features**: Minimum nights, cancellation
- **Financial Metrics**: Adjusted daily rate (ADR), revenue, availability


In [5]:
#import libraries used for the project

import numpy as np
import pandas as pd
import matplotlib as plt
import seaborn as sns
from sklearn.model_selection import train_test_split

In [10]:
# import data files
airbnb_2bed = pd.read_csv("data/airbnb_essex_2bed.csv")
airbnb_2bed.head(5)

listing_ID   town  est_rental_revenue_median  estimated_ADR  \
0  1305239269239446863  Grays                          0            NaN   
1  1323330681470281939  Grays                          0            NaN   
2   960377081653062660  Grays                          0            NaN   
3  1270246391535349924  Grays                          0            NaN   
4   660855460760807240  Grays                          0            NaN   

   Est_adj_occupancy  active_days  booking_window  length_of_stay  \
0                  0           42             NaN             NaN   
1                  0           25             NaN             NaN   
2                  0          314             NaN             NaN   
3                  0           83             NaN             NaN   
4                  0          363             NaN             NaN   

  dynamic_pricing  min_stay  ... blocked_days  rating  Num_Reviews  \
0             NaN         3  ...           11     NaN            0   
1             NaN         1  ...           25     5.0            1   
2             NaN         3  ...           52     4.0            1   
3             NaN         1  ...            0     NaN            0   
4             Low         4  ...            3     3.0            1   

   beachfront  hot_tub EV_charger kitchen  pool  pets_allowed  smoking_allowed  
0          No       No         No     Yes    No            No               No  
1          No       No         No     Yes    No            No               No  
2          No       No         No     Yes    No            No               No  
3          No       No         No     Yes    No            No               No  
4          No       No         No     Yes    No            No               No  

[5 rows x 27 columns]

In [11]:
airbnb_3bed= pd.read_csv("data/airbnb_essex_3bed.csv")
airbnb_3bed.head(5)

listing_ID             town  est_rental_revenue_median  \
0  1205963819672225857  Southend-on-sea                          0   
1   870111683833490126  Southend-on-sea                          0   
2             46934205  Southend-on-sea                          0   
3   987440215251843897  Southend-on-sea                          0   
4  1011456380483308029  Southend-on-sea                          0   

   estimated_ADR  Est_adj_occupancy  active_days  booking_window  \
0            NaN                  0          151             NaN   
1            NaN                  0          138             NaN   
2            NaN                  0          101             NaN   
3            NaN                  0          187             NaN   
4            NaN                  0           78             NaN   

   length_of_stay dynamic_pricing  min_stay  ... blocked_days  rating  \
0             NaN             NaN         5  ...            1    0.00   
1             NaN             NaN        14  ...           56    0.00   
2             NaN             NaN         1  ...          171    4.71   
3             NaN             NaN         3  ...          179    4.55   
4             NaN            High         1  ...           63     NaN   

   Num_Reviews  beachfront  hot_tub EV_charger kitchen  pool  pets_allowed  \
0            0          No       No         No     Yes    No            No   
1            0          No       No         No     Yes    No            No   
2            7          No       No         No     Yes    No            No   
3            0          No       No         No     Yes    No            No   
4            0          No       No         No     Yes    No            No   

   smoking_allowed  
0               No  
1               No  
2               No  
3               No  
4               No  

[5 rows x 27 columns]

In [13]:
#checking data of both data files

airbnb_2bed.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 879 entries, 0 to 878
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   listing_ID                 879 non-null    int64  
 1   town                       879 non-null    object 
 2   est_rental_revenue_median  879 non-null    int64  
 3   estimated_ADR              817 non-null    float64
 4   Est_adj_occupancy          879 non-null    int64  
 5   active_days                879 non-null    int64  
 6   booking_window             817 non-null    float64
 7   length_of_stay             817 non-null    float64
 8   dynamic_pricing            470 non-null    object 
 9   min_stay                   879 non-null    int64  
 10  management                 879 non-null    object 
 11  Bedrooms                   879 non-null    int64  
 12  bathrooms                  879 non-null    float64
 13  max_guests                 879 non-null    int64  

In [14]:
airbnb_3bed.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 452 entries, 0 to 451
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   listing_ID                 452 non-null    int64  
 1   town                       452 non-null    object 
 2   est_rental_revenue_median  452 non-null    int64  
 3   estimated_ADR              400 non-null    float64
 4   Est_adj_occupancy          452 non-null    int64  
 5   active_days                452 non-null    int64  
 6   booking_window             400 non-null    float64
 7   length_of_stay             400 non-null    float64
 8   dynamic_pricing            240 non-null    object 
 9   min_stay                   452 non-null    int64  
 10  management                 452 non-null    object 
 11  Bedrooms                   452 non-null    int64  
 12  bathrooms                  452 non-null    float64
 13  max_guests                 452 non-null    int64  

In [22]:
#combining both dataset: as there are the same columns, the datasets can be concatetnated. 

data = pd.concat([airbnb_2bed, airbnb_3bed], ignore_index=True)
data.shape


(1331, 27)

The dataset now as 1331 rows and 27 columns of data. 

In [26]:
data.describe()

listing_ID  est_rental_revenue_median  estimated_ADR  \
count  1.331000e+03                1331.000000    1217.000000   
mean   7.301116e+17               13002.365890     110.054207   
std    4.797764e+17               11542.536106      49.663253   
min    4.943172e+06                   0.000000      24.570000   
25%    5.362326e+07                3867.000000      82.350000   
50%    8.782748e+17               10777.000000     100.450000   
75%    1.126087e+18               19227.500000     123.750000   
max    1.375302e+18               82579.000000     719.980000   

       Est_adj_occupancy  active_days  booking_window  length_of_stay  \
count        1331.000000  1331.000000     1217.000000     1217.000000   
mean           53.491360   216.525920       39.013969       11.071487   
std            30.337682   100.565095       41.790073        8.888508   
min             0.000000    11.000000        0.000000        1.000000   
25%            30.000000   127.000000        9.000000        4.000000   
50%            60.000000   220.000000       27.000000        8.000000   
75%            78.000000   310.000000       56.000000       15.000000   
max           100.000000   366.000000      379.000000       59.000000   

          min_stay     Bedrooms    bathrooms   max_guests  listed_price  \
count  1331.000000  1331.000000  1331.000000  1331.000000   1331.000000   
mean      3.643877     2.339594     1.439895     5.019534    138.119459   
std      14.972988     0.473750     0.567074     1.360974     68.239968   
min       1.000000     2.000000     0.000000     1.000000     28.000000   
25%       1.000000     2.000000     1.000000     4.000000    100.000000   
50%       2.000000     2.000000     1.000000     5.000000    125.000000   
75%       3.000000     3.000000     2.000000     6.000000    156.000000   
max     365.000000     3.000000     4.500000    11.000000    967.000000   

       blocked_days       rating  Num_Reviews  
count   1331.000000  1179.000000  1331.000000  
mean      68.784373     4.518660    25.786627  
std       66.059811     1.015338    49.044229  
min        0.000000     0.000000     0.000000  
25%       14.000000     4.585000     1.000000  
50%       48.000000     4.850000     7.000000  
75%      108.000000     5.000000    25.000000  
max      324.000000     5.000000   431.000000

In [27]:
data.columns

Index(['listing_ID', 'town', 'est_rental_revenue_median', 'estimated_ADR',
       'Est_adj_occupancy', 'active_days', 'booking_window', 'length_of_stay',
       'dynamic_pricing', 'min_stay', 'management', 'Bedrooms', 'bathrooms',
       'max_guests', 'listed_price', 'is_active', 'economic_Category',
       'blocked_days', 'rating', 'Num_Reviews', 'beachfront', 'hot_tub',
       'EV_charger', 'kitchen', 'pool', 'pets_allowed', 'smoking_allowed'],
      dtype='object')

In [35]:
#standardising column names to lowercase
data.columns = data.columns.str.lower()

print(f"Columns in dataset :{data.columns.values}")


Columns in dataset :['listing_id' 'town' 'est_rental_revenue_median' 'estimated_adr'
 'est_adj_occupancy' 'active_days' 'booking_window' 'length_of_stay'
 'dynamic_pricing' 'min_stay' 'management' 'bedrooms' 'bathrooms'
 'max_guests' 'listed_price' 'is_active' 'economic_category'
 'blocked_days' 'rating' 'num_reviews' 'beachfront' 'hot_tub' 'ev_charger'
 'kitchen' 'pool' 'pets_allowed' 'smoking_allowed']


My target variables is occupancy rates (est_adj_occupancy) and annual rental revenue (est_rental_revenue_median). 

In [ ]:
# y - target is est_adj_occupancy and est_rental_revenue_median
y = ['estim']

In [ ]:
# split data before data wrangling and cleaning to ensure no data leakage for future testing. 
# in this instance a 70:15:15: split will be used for training:validation:testing. 